internal/distro/rhel8/imagetype.go

```diff
@@ -368,6 +371,17 @@ func overridePackageNames(packages []string) []string {
 // checkOptions checks the validity and compatibility of options and customizations for the image type.
 func (t *imageType) checkOptions(customizations *blueprint.Customizations, options distro.ImageOptions, containers []container.Spec) error {
 
+	if t.workload != nil {
+		// For now, if an image type defines its own workload, don't allow any
+		// user customizations.
+		// Soon we will have more workflows and each will define its allowed
+		// set of customizations.  The current set of customizations defined in
+		// the blueprint spec corresponds to the Custom workflow.
+		if customizations != nil {
+			return fmt.Errorf("image type %q does not support customizations", t.name)
+		}
+	}
```

Comment:
You're right that we don't use non-custom workloads in el7 and 9 but it'd be good to have the check to catch us when we change that.  For Fedora, we haven't enabled workloads on image types (yet), so I'll keep it off until we need it.

Reply:
Fair comment about Fedora. I kind of naively hope that we'll be eventually able to deduplicate most of the non-exported structures and implementations related to image type, arch and distro across various distro definitions, so that's from where my comments related to keeping them in sync come from. Keeping Fedora without workload for now seems acceptable to me.